src/Hamiltonians/Molecular.jl

```diff
@@ -186,21 +164,21 @@ function ModeIndex{2}()
     return ModeIndex{}((1, 2))
 end
 
-struct MolecularHamiltonianOffDiagonalsIterator{NA,NB}
-    address::FermiFS2C
-    op::MolecularHamiltonian
-    modes::Modes
+struct MolecularHamiltonianOffDiagonalsIterator{NA,NB,T,A<:FermiFS2C,D,OA,OB,UA,UB,TI<:FermiFSIndex}
+    address::A
+    op::MolecularHamiltonian{T,A,D}
+    modes::Modes{OA,OB,UA,UB,TI}
```

Comment:
There are now so may type parameters that the code becomes hard to read. Maybe this can be simplified, while still maintaining type stability?
```suggestion
struct MolecularHamiltonianOffDiagonalsIterator{NA,NB,T,A<:FermiFS2C,H<:MolecularHamiltonian{T,A},M<:Modes}
    address::A
    op::H
    modes::M
```
If you still want to check that `Modes` is sensible, this can be done in the type constructor.